### PR TITLE
nixosTests.glusterfs: port to python

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by flokli** [comment]
@GrahamcOfBorg test glusterfs

**Review by flokli — APPROVED** [review]

File: nixos/tests/glusterfs.nix [hunk 1]
```diff
@@ -39,27 +39,27 @@ in {
...
-    $server1->waitForUnit("glusterd.service");
-    $server2->waitForUnit("glusterd.service");
+    server1.wait_for_unit("glusterd.service")
+    server2.wait_for_unit("glusterd.service")
 
     # establish initial contact
-    $server1->succeed("sleep 2");
-    $server1->succeed("gluster peer probe server2");
-    $server1->succeed("gluster peer probe server1");
+    server1.succeed("sleep 2")
```
Comment (flokli) [inline nixos/tests/glusterfs.nix]: is there any `wait_for_open_port` we can do here instead?

**Comment by nh2** [comment]
@GrahamcOfBorg test glusterfs

File: nixos/tests/glusterfs.nix (hunk 1 above)
Comment (nh2) [reply nixos/tests/glusterfs.nix]: Fixed

**Comment by flokli** [comment]
Thanks!

**Pull request merged**